cms/djangoapps/contentstore/views/utilities/bulkupdate.py

```diff
@@ -0,0 +1,245 @@
+    """
+    Internal bulkupdate handler for POST operation
+    """
+    modified_settings = {}
+    user = request.user
+
+    # Retrieve settings from request
+    try:
+        max_attempts = request.POST.get('maxAttempts')
+        show_answer = request.POST.get('showAnswer')
+        session_status = request.session.setdefault("update_status", {})
+        session_status_string = course_key_string + max_attempts + show_answer
+    except:
+        return JsonResponse(
+            {
+                'ErrMsg': 'Missing or invalid arguments',
+                'Stage': -1
+            },
+            status=400
+        )
+
+    # Validate settings
+    _save_request_status(request, session_status_string, 1)
+
+    if max_attempts.find('null') == -1:
+        try:
+            max_attempts = int(max_attempts)
+        except:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for max attempts is not an integer',
+                    'Stage': -1
+                },
+                status=400
+            )
+        if max_attempts < 0:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for max attempts is negative',
+                    'Stage': -1
+                },
+                status=400
+            )
+        else:
+            modified_settings['max_attempts'] = max_attempts
+
+    if show_answer.find('null') == -1:
+        if show_answer in SHOW_ANSWER_OPTIONS:
+            modified_settings['showanswer'] = show_answer
+        else:
+            _save_request_status(request, session_status_string, -1)
+            return JsonResponse(
+                {
+                    'ErrMsg': 'Given value for show answer is not an available option',
+                    'Stage': -1
+                },
+                status=400
+            )
+    _save_request_status(request, session_status_string, 2)
+    _update_advanced_settings(request, course_key_string, session_status_string, modified_settings)
+    _save_request_status(request, session_status_string, 3)
+    bulk_update_problem_settings.delay(course_key_string, user.id, user.username, user.email, modified_settings)
+
+    return JsonResponse({'Status': 'OK'})
+
+
+@login_required
+def utility_bulkupdate_handler(request, course_key_string):
+    """
+    Handler for bulk update requests in the utilities tool. Currently
+    updates maxAttempts and showAnswer settings for all problems in a
+    given course and sets as settings for future problems in advanced
+    settings
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    if not has_course_author_access(request.user, course_key):
+        raise PermissionDenied()
+
+    if 'text/html' in request.META.get('HTTP_ACCEPT', 'text/html'):
```

Comment:
Why the `in` check instead of `==`?

Reply:
The call to `request.META.get('HTTP_ACCEPT', 'text/html')` returns multiple types it can accept as a string, e.g. `'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8'`